File: Algoritmo(C)/Alg_S8_6.c
#include <stdio.h>
int main(){
    //variable
    float vetor[5];
    int i, cod;
    //input
    printf("0 - sair\n");
    printf("1 - ordem direta\n");
    printf("2 - ordem inversa\n");
    printf("Digite o codigo: ");
    scanf("%d",&cod);
    while((cod < 0) || (cod > 2)){
        printf("Codigo invalido\n");
        printf("Digite o codigo: ");
        scanf("%d",&cod);
    }
    //input + process
    while(cod != 0){
        for(i = 0; i <= 4; i++){
            printf("Digite um valor para o vetor %d/5: ",(i + 1));
            scanf("%f",&vetor[i]);
        }
        //output
        switch(cod){
            case (1):{
                for(i = 0; i <= 4; i++){
                    printf("Vetor[%d] = %.2f\n",(i + 1),vetor[i]);
                }
                break;
            }
            case (2):{
                for(i = 4; i >= 0; i--){
                    printf("Vetor[%d] = %.2f\n",(i + 1),vetor[i]);
                }
                break;
            }
        }
        //input
        printf("0 - sair\n");
        printf("1 - ordem direta\n");
        printf("2 - ordem inversa\n");
        printf("Digite o codigo: ");
        scanf("%d",&cod);
        while((cod < 0) || (cod > 2)){
            printf("Codigo invalido\n");
            printf("Digite o codigo: ");
            scanf("%d",&cod);
        }
    }
}

File: Algoritmo(Python)/Alg_S8_6.py
#variables
vetor = []
#input
print("0 - Finaliza")
print("1 - Ordem direta")
print("2 - Ordem inversa")
cod = int(input("Digite um código: "))
if(cod != 0):
    while((cod < 0) or (cod > 2)):
        print("Código inválido")
        cod = int(input("Digite um código: "))
    for(i) in range(0, 5):
        num = float(input("Digite um número: "))
        vetor.append(num)
    if(cod == 1):
        for(i) in (vetor):
            print(i)
    else:
        vetor.reverse()
        for(i) in (vetor):
            print(i)